### PR TITLE
IGNITE-15427 Improve the snapshot procedure logging

### PR DESCRIPTION
**Pull request by Mmuzaf**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java [hunk 1]
```diff
@@ -683,6 +693,20 @@ private Runnable wrapExceptionIfStarted(IgniteThrowableRunner exec) {
         return true;
     }
 
+    /**
+     * @param grps List of processing pairs.
+     * @return Map of cache group id their partitions compacted by {@link S#compact(Collection)}.
+     */
+    private static Map<Integer, String> convert(Collection<GroupPartitionId> grps) {
```
Comment (xtern) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java:700]: compactGroupPartitions?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java [hunk 2]
```diff
@@ -1420,7 +1420,14 @@ public static boolean isSnapshotOperation(DiscoveryEvent evt) {
 
             // Schedule task on a checkpoint and wait when it starts.
             try {
+                long start = U.currentTimeMillis();
+
                 task.started().get();
+
+                if (log.isInfoEnabled()) {
+                    log.info("Finished waiting for synchronized checkpoint under topology lock " +
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java [hunk 3]
```diff
@@ -1420,7 +1420,14 @@ public static boolean isSnapshotOperation(DiscoveryEvent evt) {
...
+                        "[snapName=" + task.snapshotName() + ", time=" + (U.currentTimeMillis() - start) + "ms]");
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java [hunk 4]
```diff
@@ -394,7 +394,14 @@ public boolean start() {
 
         if (withMetaStorage) {
             try {
+                long start = U.currentTimeMillis();
+
                 U.get(((DistributedMetaStorageImpl)cctx.kernalContext().distributedMetastorage()).flush());
+
+                if (log.isInfoEnabled()) {
+                    log.info("Finished waiting for all the concurrent operations over the metadata store before snapshot " +
+                        "[snapName=" + snpName + ", time=" + (U.currentTimeMillis() - start) + "ms]");
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java [hunk 5]
```diff
@@ -518,8 +525,11 @@ else if (affNode && missed.isEmpty() && cctx.kernalContext().query().moduleEnabl
         // Submit all tasks for partitions and deltas processing.
         List<CompletableFuture<Void>> futs = new ArrayList<>();
 
-        if (log.isInfoEnabled())
-            log.info("Submit partition processing tasks with partition allocated lengths: " + partFileLengths);
+        if (log.isInfoEnabled()) {
+            log.info("Submit partition processing tasks to the snapshot execution pool " +
+                "[map=" + convert(partFileLengths.keySet()) +
+                ", totalSizes=" + U.humanReadableByteCount(partFileLengths.values().stream().mapToLong(v -> v).sum()) + ']');
```
[the remark this answers is not in the document]
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java (hunk 1 above)
Comment (Mmuzaf) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotFutureTask.java]: Fixed.

**Pull request merged**